unity/Assets/Scripts/AgentManager.cs

```diff
@@ -1295,31 +1295,53 @@ bool shouldRenderImageSynthesis
+                cMetadata.worldRelativeThirdPartyCameraRotation = camera.transform.eulerAngles;
+
+                //grab this for parent or agent relative stuff
+                var worldSpaceCameraRotationAsQuaternion = camera.transform.rotation;
+
+                //this may be the same info as the agentPositionRelative values since ThirdPartyCameras often are children of the agent
+                //but in cases where a third party camera is attached to an arm joint or the like, this can be useful
+                if (camera.transform.parent != null) {
+                    cMetadata.parentObjectName = camera.transform.parent.name;
+
+                    cMetadata.parentPositionRelativeThirdPartyCameraPosition =
+                        camera.transform.parent.InverseTransformPoint(camera.transform.position);
+
+                    //get third party camera rotation as quaternion in world space
+                    var parentSpaceCameraRotationAsQuaternion =
+                        Quaternion.Inverse(camera.transform.parent.rotation)
+                        * worldSpaceCameraRotationAsQuaternion;
+                    cMetadata.parentPositionRelativeThirdPartyCameraRotation =
+                        parentSpaceCameraRotationAsQuaternion.eulerAngles;
                 } else {
-                    //if this third party camera is not a child of the agent, then the agent relative coordinates
-                    //are the same as the world coordinates so
-                    cMetadata.agentPositionRelativeThirdPartyCameraPosition = camera
-                        .gameObject
-                        .transform
-                        .position;
-                    cMetadata.agentPositionRelativeThirdPartyCameraRotation = camera
-                        .gameObject
-                        .transform
-                        .eulerAngles;
+                    cMetadata.parentObjectName = "";
+                    cMetadata.parentPositionRelativeThirdPartyCameraPosition = null;
+                    cMetadata.parentPositionRelativeThirdPartyCameraRotation = null;
+                }
+
+                //if this camera is part of the agent's heirarchy at all, get agent relative info
+                if (camera.GetComponentInParent<BaseAgentComponent>() != null) {
+                    GameObject agent = camera.GetComponentInParent<BaseAgentComponent>().gameObject;
+
+                    cMetadata.agentPositionRelativeThirdPartyCameraPosition =
+                        agent.transform.InverseTransformPoint(camera.gameObject.transform.position);
+
+                    //get third party camera rotation as quaternion in world space
+                    var agentSpaceCameraRotationAsQuaternion =
+                        Quaternion.Inverse(agent.transform.rotation)
+                        * worldSpaceCameraRotationAsQuaternion;
+                    cMetadata.agentPositionRelativeThirdPartyCameraRotation =
```

Comment:
Change variable name to `agentRotationRelativeThirdPartyCameraRotation`

Reply:
updated in ef3bfb20c08ec9a7a991b23d7e9008af3c72d132